### ISSUE TITLE
False-positive MixedMethodCall if a var wasn't defined on top level

### ISSUE THREAD

**Issue by fluffycondor** [vimeo/psalm#6558]
https://psalm.dev/r/f0288c0f34
Expected: no errors.
Got: MixedMethodCall

Defining the var fixes the problem:
https://psalm.dev/r/369beda054

**Comment by orklah** [comment]
Would you have expected to get another issue? Because it makes sense for Psalm to warn you there.

**Comment by weirdan** [comment]
```php
$foo = true;
require 'Foo.php';
```
will cause random failures, so it indeed makes sense to flag this.

**Comment by orklah** [comment]
Psalm reports the same issue when in a closed scope however
